### PR TITLE
Fixes #982. Upgrade Intern to 3.4.0

### LINKED ISSUE #982: Update Intern to 3.4.0
The main reason I think we should do this is for the new `bail` option:

https://github.com/theintern/intern/commit/a6a732ef4864f23f68aee3bbb4ebb27c888f0992

What that means is if we hit a single test failure, Intern doesn't bother to run any more tests.

The latest release is 3.0.6: https://github.com/theintern/intern/releases/tag/3.0.6

So we'll probably want whatever comes next.

### PR DESCRIPTION
**Pull request by miketaylr**
- [x] get it working
- [x] update docs
- [x] update travis
- [x] fix failing tests

Not ready for review yet.

### CODE REVIEW

**Comment by miketaylr** [comment]
FYI @karlcow, this is working on my machine™, so you might wait until this is merged to not have to downgrade libs or browsers to run tests locally.

**Comment by karlcow** [comment]
❤️ Thanks

**Comment by miketaylr** [comment]
It seems like there's a few tests that are broken now, as a result of changes in either how Intern is working, or Selenium or something. Will try to fix these ASAP and get this merged, this is a major win in terms of simplifying dev environment setup.

**Comment by miketaylr** [comment]
Locally:

```
Fx 50 MAC:  [✓✓✓✓✓✓✓✓✓✓✓×✓✓✓×✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓×✓✓✓✓✓] 68/68, 3 fail
```

Getting there.

**Comment by miketaylr** [comment]
Hmm... `18/68 tests failed`. There's some timing differences it seems that I don't understand w/ Intern 3. Things are green locally, but the travis linux VMs either timeout or race.

**Comment by miketaylr** [comment]
https://github.com/webcompat/webcompat.com/pull/1277/commits/43e593d42f3e8cc8045c8ee9107dfe8d23c252db is dumb. But...

`5/68 tests failed`

There seems to be different behavior when Intern commands "cross" page loads. Like they execute too quickly, or something.

**Comment by miketaylr** [comment]
Getting closer... though obviously a bit racy still.

screen shot 2017-01-14 at 11 54 27 pm

**Comment by magsout** [comment]
@miketaylr 👍

**Comment by miketaylr** [comment]
Sigh, still super intermittent. Failing on some auth tests, and not others... then different ones when re-trying the build.

**Comment by miketaylr** [comment]
Hope to figure this out this week.

**Comment by miketaylr** [comment]
Whoa, push and pr run passed. Will re-trigger to tempt my luck... 💣

**Comment by karlcow** [comment]
This is good news.

**Comment by miketaylr** [comment]
> This is good news.

Hopefully! I want to unblock you ASAP. It seems like *maybe* we're running into a geckodriver/marionette limitation (https://bugzilla.mozilla.org/show_bug.cgi?id=1291320, https://github.com/mozilla/geckodriver/issues/308), which is why the regular login stuff that used to work in Intern 2 is *sometimes* failing w/ Intern 3.

**Comment by miketaylr** [comment]
OK, this is too flaky. @vladikoff said he would help me debug when we're in Mountain View next week.

**Comment by miketaylr** [comment]
> OK, this is too flaky. @vladikoff said he would help me debug when we're in Mountain View next week.

We found out there was a bug in my login method that didn't surface w/ Intern 2 -- if we get the extra confirmation page from GitHub, you have to click on a button. In Intern 3, it's clicking on it *sometimes* too quickly (so possibly we were clicking on the original submit twice, rather than once on submit, then once on confirmation). 

Also we have a theory that if you click on a button *too* quickly, GitHub thinks you're a bot. So, we added a small timeout if you see that extra confirmation page.

Anyways, let's see if the tests actually pass now for both runs.

**Comment by miketaylr** [comment]
> All checks have passed

Let's retrigger.

**Comment by magsout** [comment]
>Also we have a theory that if you click on a button *too* quickly, GitHub thinks you're a bot. So, we added a small timeout if you see that extra confirmation page.


Good catch... hard to find this bug...

**Comment by vladikoff** [comment]
@magsout hey man, how is it going!? 🥇

File: tests/functional/comments-auth.js [hunk 1]
```diff
@@ -27,26 +27,16 @@ define([
...
     'Comments form visible when logged in': function() {
-      return this.remote
-        .setFindTimeout(intern.config.wc.pageLoadTimeout)
-        .get(require.toUrl(url('/issues/100')))
-        .then(FunctionalHelpers.visibleByQSA('.js-Comment-form'))
-        .findByCssSelector('.js-Comment-form').isDisplayed()
-        .then(function(isDisplayed) {
-          assert.equal(isDisplayed, true, 'Comment form visible for logged in users.');
-        })
+      return FunctionalHelpers.openPage(this, url('/issues/100'), '.js-Issue')
```
Comment (karlcow) [inline tests/functional/comments-auth.js]: wow nice that simplify things a lot.

File: tests/functional/image-uploads-non-auth.js [hunk 2]
```diff
@@ -2,26 +2,25 @@
...
   'require',
-  'intern/node_modules/dojo/node!path',
-], function(intern, registerSuite, assert) {
+  'tests/functional/lib/helpers',
+  'intern/browser_modules/dojo/node!path'
+], function(intern, registerSuite, assert, require, FunctionalHelpers) {
   'use strict';
 
-  var url = intern.config.siteRoot;
+  var url = intern.config.siteRoot + '/?open=1';
```
Comment (karlcow) [inline tests/functional/image-uploads-non-auth.js]: Should we switch to the /issues/new URI?

**Comment by karlcow** [comment]
And given that github markdown strips the style attribute let's roll it. 🎢

**Comment by magsout** [comment]
@vladikoff hey, fine and you ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -210,11 +210,11 @@ For testing code locally, you will need a very basic setup. There are a few requ
...
-
-Windows typically doesn't have the *make* tool installed. Windows users without *make* should look at the "detailed setup" section below.
-
-As an alternative to Windows, a cloud IDE such as [Cloud 9](https://c9.io) can be used for a relatively easier setup. If you take this route, please update to the latest Python version with the following. (This is to avoid `InsecurePlatformWarning` errors that arise when the default Python 2.7.6 is used).
+> Note: If you install Python on Windows using the MSI installer, it is highly recommended to check the "Add to path"-box during installation. If you have not done so, see if one of the answers to the StackOverflow post [Adding Python path on Windows 7](http://stackoverflow.com/questions/6318156/adding-python-path-on-windows-7) can help you - it should also work fine for later versions of Windows.
```
Comment (miketaylr) [inline CONTRIBUTING.md]: Ah, yeah. I just thought it looked cool visually -- happy to merge in a PR if you'd like to change it back. :)

**Comment by miketaylr** [comment]
Thanks @karlcow @vladikoff!

**Pull request merged**